test/unittests/clients/test_group_user_clients_live.py

```diff
@@ -0,0 +1,149 @@
+                    "name": "testgroup",
+                    "descriptions": [],
+                    "project": {
+                        "id": PROJECT_IRI,
+                        "shortname": "systematic-tp",
+                    },
+                },
+                {
+                    "id": "http://rdfh.ch/groups/4123/anotherGroupId",
+                    "name": "testgroupReaders",
+                    "descriptions": [],
+                    "project": {
+                        "id": "http://rdfh.ch/projects/anotherProject",
+                        "shortname": "another-project",
+                    },
+                },
+            ]
+        }
+        with patch("dsp_tools.clients.group_user_clients_live.requests.get", return_value=mock_response) as mock_get:
+            result = group_client.get_all_groups()
+        assert len(result) == 2
+        assert result[0]["name"] == "testgroup"
+        assert result[0]["id"] == GROUP_IRI
+        assert result[1]["name"] == "testgroupReaders"
+        assert mock_get.call_args[0][0] == f"{group_client.api_url}/admin/groups"
+
+    def test_get_all_groups_empty(self, group_client: GroupClientLive) -> None:
+        mock_response = Mock(status_code=200, ok=True, headers={})
+        mock_response.json.return_value = {"groups": []}
+        with patch("dsp_tools.clients.group_user_clients_live.requests.get", return_value=mock_response):
+            result = group_client.get_all_groups()
+        assert result == []
+
+    def test_get_all_groups_timeout(self, group_client: GroupClientLive) -> None:
+        with patch(
+            "dsp_tools.clients.group_user_clients_live.requests.get", side_effect=requests.ReadTimeout("Timeout")
+        ):
+            with pytest.raises(DspToolsRequestException):
+                group_client.get_all_groups()
+
+    def test_get_all_groups_server_error(self, group_client: GroupClientLive) -> None:
+        mock_response = Mock(status_code=500, ok=False, headers={}, text="Internal Server Error")
+        mock_response.json.return_value = {}
+        with patch("dsp_tools.clients.group_user_clients_live.requests.get", return_value=mock_response):
+            with pytest.warns(DspToolsUnexpectedStatusCodeWarning):
+                result = group_client.get_all_groups()
+        assert result == []
+
+
+class TestGroupClientCreateNewGroup:
+    """response
+
+        {
+        "group": {
+            "id": "http://rdfh.ch/groups/9999/krZ2lP_NTBSbfbX2BSvhiw",
+            "name": "NewGroup",
+            "descriptions": [
+                {
+                    "value": "NewGroupDescription",
+                    "language": "en"
+                },
+                {
+                    "value": "NeueGruppenBeschreibung",
+                    "language": "de"
+                }
+            ],
+            "project": {
+                "id": "http://rdfh.ch/projects/zimXRcPvRxeXS-6veIrB7A"
+        }
+    }
+    """
```

Comment:
First I was puzzled, because this JSON is invalid (a closing bracket is missing). Then I realised it's only a docstring, so it's unused.

Was this just a temporary note that can be deleted now? --> Delete it.

Or do you plan to use it later? --> fix it:


```suggestion
    """response
    {
        "group": {
            "id": "http://rdfh.ch/groups/9999/krZ2lP_NTBSbfbX2BSvhiw",
            "name": "NewGroup",
            "descriptions": [
                {
                    "value": "NewGroupDescription",
                    "language": "en"
                },
                {
                    "value": "NeueGruppenBeschreibung",
                    "language": "de"
                }
            ],
            "project": {
                "id": "http://rdfh.ch/projects/zimXRcPvRxeXS-6veIrB7A"
            }
        }
    }
    """
```